### PR TITLE
feat: add digest authentication

### PR DESCRIPTION
**Pull request by Jakub-Vacek**
This PR adds basic description of Digest Auth in provider.json. Waiting on actually released digest in sdk.

### CODE REVIEW

File: docs/guides/add-new-provider.md [hunk 1]
```diff
@@ -221,6 +222,41 @@ _e.g. `https://api.example.com/?accessKey=<apikey>`_
 - `in` must be set to `query`
 - `name` is the query param name that holds the API key (e.g. `accessKey`)
 
+### Digest Auth {#digest-auth}
+
+_`Authorization: Digest <credentials>` in headers_
```
Comment (kysely) [inline docs/guides/add-new-provider.md:227]:
```suggestion
_e.g. `Authorization: Digest <credentials>` in headers_
```

**Review by kysely — COMMENTED** [review]
👍 I honestly don't have any comments that would improve your work here.

Just a reminder we have this information in other places in docs so it might be worth updating those, too:
- [Reference/Super.json](https://superface.ai/docs/reference/superjson#securityvalues)
- [Comlink/Reference/Provider](https://superface.ai/docs/comlink/reference/provider#security-scheme)

**Comment by jnv** [comment]
@Jakub-Vacek What is the status, has this been released?

**Comment by Jakub-Vacek** [comment]
It hasn't 😒 We have it in release candidate `1.3.0-rc.0 `

**Comment by Jakub-Vacek** [comment]
@jnv Digest in SDK is released (1.3.0), this can be merged.

File: docs/comlink/reference/provider.mdx [hunk 2]
```diff
@@ -115,6 +115,28 @@ This security scheme refers to HTTP Basic Authentication as described in [RFC 72
...
 ```
 
+#### Digest Auth
+
+This security scheme refers to HTTP Digest Authentication as described in [RFC 7616](https://datatracker.ietf.org/doc/html/rfc7616).
+
+- `id` (string) - Unique ID for security scheme
+- `type`: "http" (string)
+- `scheme`: "digest" (string)
+- `statusCode` (number, optional) - HTTP status code that should be returned from initial call (containing challenge header). Default value is `401`.
```
Comment (jnv) [inline docs/comlink/reference/provider.mdx:125]: I don't quite understand the description, let me try to rephrase it.

```suggestion
- `statusCode` (number, optional) - HTTP status code expected for the initial response with the challenge header. Default value is `401`.
```

File: docs/comlink/reference/provider.mdx [hunk 3]
```diff
@@ -115,6 +115,28 @@ This security scheme refers to HTTP Basic Authentication as described in [RFC 72
...
+- `challengeHeader` (string, optional) - Case insensitive name of header containing challenge from the server. Default value is `www-authenticate`.
```
Comment (jnv) [inline docs/comlink/reference/provider.mdx:126]:
```suggestion
- `challengeHeader` (string, optional) - Case insensitive name of the response header with server issued challenge. Default value is [`www-authenticate`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/WWW-Authenticate).
```

File: docs/comlink/reference/provider.mdx [hunk 4]
```diff
@@ -115,6 +115,28 @@ This security scheme refers to HTTP Basic Authentication as described in [RFC 72
...
+- `authorizationHeader` (string, optional) - Case insensitive name of header containing authorization. Default value is `Authorization`.
```
Comment (jnv) [inline docs/comlink/reference/provider.mdx:127]:
```suggestion
- `authorizationHeader` (string, optional) - Case insensitive name of the request header containing digest authorization details. Default value is [`authorization`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization).
```

File: docs/guides/add-new-provider.md [hunk 5]
```diff
@@ -221,6 +222,41 @@ _e.g. `https://api.example.com/?accessKey=<apikey>`_
...
+  ]
+}
+```
+
+_Replace the security scheme `id` value in the example with your own ID._
+
+#### Optional parameters:
+- `statusCode` HTTP status code that should be returned from initial call (containing challenge header). Default value is `401`.
+- `challengeHeader` Case insensitive name of header containing challenge from the server. Default value is `www-authenticate`.
+- `authorizationHeader` Case insensitive name of header containing authorization. Default value is `Authorization`.
```
Comment (jnv) [inline docs/guides/add-new-provider.md:256]: I would remove this section and added a link to the relevant reference section instead. I suppose these parameters will be rarely needed.

**Review by jnv — CHANGES_REQUESTED** [review]

**Comment by jnv** [comment]
@Jakub-Vacek Since the feature was released, could you please revise and merge this?

**Review by jnv — APPROVED** [review]
:shipit:

**Comment by Jakub-Vacek** [comment]
@jnv I am sorry I completely forgot about this PR for some reason 🤦🏻‍♂️😩 I have resolved conflicts and I think it is good to go. Can you please check if it still ok? 🙏

**Review by jnv — APPROVED** [review]
LGTM! :shipit:

**Pull request merged**